src/lib/components/Footer/FooterIcon.tsx

```diff
@@ -0,0 +1,16 @@
+import { FC, PropsWithChildren, ComponentProps } from 'react';
+import classNames from 'classnames';
+
+export type FooterIconProps = PropsWithChildren<{
+  href?: string;
+  className?: string;
+  icon?: FC<ComponentProps<'svg'>>;
+}>;
+
+export const FooterIcon: FC<FooterIconProps> = ({ href, className, icon: Icon }) => {
+  return (
+    <a href={href} className={classNames('text-gray-500  dark:hover:text-white', className)}>
```

Comment:
Links should have an accessible label. This link will read nothing on a screen reader because it has no text inside

Reply:
also, since `href` is optional we need to do some control here to not display it as a link, or by making the `href` required.